open-craft/edx-platform
# BB-986: Add mktg links override capability

This PR enables us to set a configuration variable or site configuration to override any marketing link with a valid URL.

**Testing instructions:**
1. Pull this branch on you local devstack.
2. Run tests on LMS shell with `pytest common/djangoapps/edxmako/tests.py::ShortcutsTests`
3. Add `MKTG_URL_OVERRIDES` on `lms.env.json`:
```
....
"MKTG_URL_OVERRIDES": {
    "TOS": "https://www.edx.org"
},
...
```
4. Restart LMS and memcached (`make lms-restart && docker-compose restart memcached`) and check if the Terms of service link redirects to www.edx.org.
5. Change the override to a invalid URL (like `12345`) and check that the link disappears.
6. Redo steps 3..5 but setting `MKTG_URL_OVERRIDES` on `SiteConfiguration` instead of in `lms.env.json`.

**Reviewers:**
